pkg/config/kds/config.go

```diff
@@ -9,32 +9,81 @@ import (
 	"github.com/Kong/kuma/pkg/config"
 )
 
-func DefaultKumaDiscoveryServerConfig() *KumaDiscoveryServerConfig {
-	return &KumaDiscoveryServerConfig{
-		GrpcPort:        5685,
-		RefreshInterval: 1 * time.Second,
+func DefaultKdsConfig() *KdsConfig {
+	return &KdsConfig{
+		Server: &KdsServerConfig{
+			GrpcPort:        5685,
+			RefreshInterval: 1 * time.Second,
+		},
+		Client: &KdsClientConfig{},
 	}
 }
 
-// Kuma Discovery Service (KDS) server configuration.
-type KumaDiscoveryServerConfig struct {
+// Kuma Discovery Service (KDS) configuration.
+type KdsConfig struct {
+	// Server stores configuration for the KDS server part.
+	Server *KdsServerConfig `yaml:"server"`
+	// Client stores configuration for the KDS client part.
+	Client *KdsClientConfig `yaml:"client"`
+}
+
+var _ config.Config = &KdsConfig{}
+
+func (c *KdsConfig) Sanitize() {
+}
+
+func (c *KdsConfig) Validate() (errs error) {
+	if err := c.Server.Validate(); err != nil {
+		return errors.Wrap(err, "Server validation failed")
+	}
+	if err := c.Client.Validate(); err != nil {
+		return errors.Wrap(err, "Client validation failed")
+	}
+	return nil
+}
+
+type KdsServerConfig struct {
 	// Port of a gRPC server that serves Kuma Discovery Service (KDS).
-	GrpcPort uint32 `yaml:"grpcPort" envconfig:"kuma_discovery_server_grpc_port"`
+	GrpcPort uint32 `yaml:"grpcPort" envconfig:"kuma_kds_server_grpc_port"`
 	// Interval for refreshing state of the world
-	RefreshInterval time.Duration `yaml:"refreshInterval" envconfig:"kuma_discovery_server_refresh_interval"`
+	RefreshInterval time.Duration `yaml:"refreshInterval" envconfig:"kuma_kds_server_refresh_interval"`
+	// TlsCertFile defines a path to a file with PEM-encoded TLS cert.
+	TlsCertFile string `yaml:"tlsCertFile" envconfig:"kuma_kds_server_tls_cert_file"`
+	// TlsKeyFile defines a path to a file with PEM-encoded TLS key.
+	TlsKeyFile string `yaml:"tlsKeyFile" envconfig:"kuma_kds_server_tls_key_file"`
 }
 
-var _ config.Config = &KumaDiscoveryServerConfig{}
+var _ config.Config = &KdsServerConfig{}
 
-func (c *KumaDiscoveryServerConfig) Sanitize() {
+func (c *KdsServerConfig) Sanitize() {
```

Comment:
Should TlsKeyFile be sanitized?

Reply:
It's just a path. I think it's fine to have it unsanitized. You can't see the content of this.